### ISSUE TITLE
error when compiling collective pallet by itself

### ISSUE THREAD

**Issue by bddap** [paritytech/substrate#6450]
I found the issue when trying to use pallet-collective as a dependency, but the compile error can also be reproduced like so:

```
cd frame/collective
cargo check --no-default-features
```

Result

```
    Checking pallet-collective v2.0.0-rc3 (/home/a/d/substrate/frame/collective)                         
error: cannot find macro `vec` in this scope                                                             
   --> frame/collective/src/lib.rs:535:49                                                                
    |                                                                                                    
535 |                 let votes = Votes { index, threshold, ayes: vec![who.clone()], nays: vec![], end };
    |                                                             ^^^                                    
                                                                                                         
error: cannot find macro `vec` in this scope                                                             
   --> frame/collective/src/lib.rs:535:74                                                                
    |                                                                                                    
535 |                 let votes = Votes { index, threshold, ayes: vec![who.clone()], nays: vec![], end };
    |                                                                                      ^^^           
                                                                                                         
error: aborting due to 2 previous errors                                                                 
                                                                                                         
error: could not compile `pallet-collective`.                                                            
                                                                                                         
To learn more, run the command again with --verbose.                                                     
```

## Quick fix

in `./frame/collective/src/lib.rs` change

```rust
use sp_std::{prelude::*, result};
```

to

```rust
use sp_std::{prelude::*, result, vec};
```

## Suspected cause

`sp_std::prelude` apparently tries to export `vec!`, but it doesn't seem to work when imported as part of the glob. It's possible that `sp_std::prelude` is exporting the `alloc::vec` module but not the `vec!` macro.

**Comment by bddap** [comment]
PS: that suspected cause is just a guess

**Comment by bkchr** [comment]
So, the problem is that we have `pallet-balances` in `dev-dependencies` and this makes `std` being leaked, aka `std` feature is enabled. And on `std` we don't have `vec!` in the `sp-std` prelude, but `no_std` is enabled and this also removed the default `vec!` import.

TLDR: Yes this does not work, but there is also no solution for now. Sorry!

**Comment by garrettmaring** [comment]
I am running into this at the moment when building a project with 2 sibling contracts

**Comment by 99Kies** [comment]
I had the same problem when I used `cargo build --release -p xxxx-runtime`, but I was able to run `cargo build --release` successfully.

```
$ cargo build --release -p dorafactory-node-runtime
......
......
  Compiling parachain-info v0.1.0 (https://github.com/paritytech/cumulus?branch=polkadot-v0.9.17#db11baac)
error: cannot find macro `vec` in this scope
   --> /Users/feng/.cargo/git/checkouts/substrate-7e08433d4c370a21/22d40c7/frame/collective/src/lib.rs:556:38
    |
556 |                     Votes { index, threshold, ayes: vec![], nays: vec![], end }
    |                                                     ^^^
    |
    = note: consider importing one of these items:
            codec::alloc::vec
            scale_info::prelude::vec
            sp_std::vec

error: cannot find macro `vec` in this scope
   --> /Users/feng/.cargo/git/checkouts/substrate-7e08433d4c370a21/22d40c7/frame/collective/src/lib.rs:556:52
    |
556 |                     Votes { index, threshold, ayes: vec![], nays: vec![], end }
    |                                                                   ^^^
    |
    = note: consider importing one of these items:
            codec::alloc::vec
            scale_info::prelude::vec
            sp_std::vec

error: cannot find macro `vec` in this scope
    --> /Users/feng/.cargo/git/checkouts/substrate-7e08433d4c370a21/22d40c7/frame/society/src/lib.rs:1815:4
     |
1815 |             vec![]
     |             ^^^
     |
     = note: consider importing one of these items:
             codec::alloc::vec
             scale_info::prelude::vec
             sp_std::vec

error: cannot find macro `vec` in this scope
   --> /Users/feng/.cargo/git/checkouts/substrate-7e08433d4c370a21/22d40c7/frame/society/src/lib.rs:618:22
    |
618 |             let mut members = vec![];
    |                               ^^^
    |
    = note: consider importing one of these items:
            codec::alloc::vec
            scale_info::prelude::vec
            sp_std::vec

error: could not compile `pallet-collective` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by bkchr** [comment]
@99Kies this very probably means that you don't have the `std` feature enabled for these pallets in your runtime Cargo.toml
